loadbalancer/algorithm.go

```diff
@@ -119,6 +120,7 @@ func withFadeIn(rnd *rand.Rand, ctx *routing.LBContext, notFadingIndexes []int,
 	if rnd.Float64() < f {
 		return ep[choice]
 	}
+	notFadingIndexes := []int{}
```

Comment:
Here we can pre-allocate len(ep) capacity

Reply:
https://github.com/zalando/skipper/pull/2412#discussion_r1238267164

Reply:
I think we should not preallocate here.

Actually, `notFadingIndexes` length depends on the number of non-fading endpoints which may change over time. So, capacity to allocate with is not known.
However, I tried to preallocate with `make([]int, 0, len(ep))` here and tested the performance, I see it's degrading here:

```
                                           │ lockedRand.txt │           preallocate2.txt           │
                                           │     sec/op     │    sec/op     vs base                │
FadeIn/random,_10,_1_clients-16                894.0n ±  5%   1619.0n ± 5%  +81.10% (p=0.000 n=26)
FadeIn/random,_11,_1_clients-16                403.8n ±  9%    417.8n ± 8%        ~ (p=0.304 n=26)
FadeIn/random,_10,_4_clients-16                879.0n ± 12%   1031.5n ± 6%  +17.34% (p=0.000 n=26)
FadeIn/random,_11,_4_clients-16                548.4n ±  7%    592.6n ± 6%        ~ (p=0.102 n=26)
FadeIn/random,_10,_16_clients-16               914.6n ±  3%   1101.0n ± 2%  +20.38% (p=0.000 n=26)
FadeIn/random,_11,_16_clients-16               624.3n ±  6%    621.0n ± 5%        ~ (p=0.710 n=26)
FadeIn/random,_10,_64_clients-16               1.003µ ±  3%    1.183µ ± 1%  +18.00% (p=0.000 n=26)
FadeIn/random,_11,_64_clients-16               700.5n ±  4%    704.1n ± 6%        ~ (p=0.498 n=26)
FadeIn/random,_10,_256_clients-16              1.016µ ±  4%    1.156µ ± 1%  +13.84% (p=0.000 n=26)
FadeIn/random,_11,_256_clients-16              703.8n ±  2%    704.0n ± 5%        ~ (p=0.770 n=26)
FadeIn/round-robin,_10,_1_clients-16           935.7n ±  5%   1704.5n ± 4%  +82.16% (p=0.000 n=26)
FadeIn/round-robin,_11,_1_clients-16           383.0n ±  6%    432.8n ± 5%  +13.00% (p=0.000 n=26)
FadeIn/round-robin,_10,_4_clients-16           629.8n ± 10%    905.1n ± 5%  +43.72% (p=0.000 n=26)
FadeIn/round-robin,_11,_4_clients-16           341.4n ± 17%    379.7n ± 8%  +11.23% (p=0.047 n=26)
FadeIn/round-robin,_10,_16_clients-16          624.9n ±  7%    775.5n ± 4%  +24.10% (p=0.000 n=26)
FadeIn/round-robin,_11,_16_clients-16          350.7n ±  7%    377.6n ± 8%        ~ (p=0.069 n=26)
FadeIn/round-robin,_10,_64_clients-16          650.5n ±  3%    819.0n ± 3%  +25.91% (p=0.000 n=26)
FadeIn/round-robin,_11,_64_clients-16          383.2n ±  1%    384.4n ± 6%        ~ (p=0.840 n=26)
FadeIn/round-robin,_10,_256_clients-16         651.6n ±  1%    815.8n ± 2%  +25.21% (p=0.000 n=26)
FadeIn/round-robin,_11,_256_clients-16         386.9n ±  1%    390.6n ± 2%        ~ (p=0.161 n=26)
FadeIn/consistent-hash,_10,_1_clients-16       1.185µ ±  5%    2.211µ ± 3%  +86.54% (p=0.000 n=26)
FadeIn/consistent-hash,_11,_1_clients-16       627.8n ±  9%    663.2n ± 6%        ~ (p=0.234 n=26)
FadeIn/consistent-hash,_10,_4_clients-16       568.9n ±  4%    873.6n ± 4%  +53.55% (p=0.000 n=26)
FadeIn/consistent-hash,_11,_4_clients-16       452.2n ±  8%    433.4n ± 8%        ~ (p=0.426 n=26)
FadeIn/consistent-hash,_10,_16_clients-16      462.2n ±  9%    598.2n ± 2%  +29.41% (p=0.000 n=26)
FadeIn/consistent-hash,_11,_16_clients-16      430.4n ±  5%    429.6n ± 5%        ~ (p=0.602 n=26)
FadeIn/consistent-hash,_10,_64_clients-16      483.9n ±  1%    636.8n ± 4%  +31.60% (p=0.000 n=26)
FadeIn/consistent-hash,_11,_64_clients-16      430.9n ±  1%    428.9n ± 5%        ~ (p=0.558 n=26)
FadeIn/consistent-hash,_10,_256_clients-16     497.3n ±  2%    629.1n ± 2%  +26.50% (p=0.000 n=26)
FadeIn/consistent-hash,_11,_256_clients-16     433.8n ±  5%    433.4n ± 7%        ~ (p=0.680 n=26)
geomean                                        582.4n          690.8n       +18.61%
```